src/AutoRest.CSharp/Common/AutoRest/Plugins/CSharpGen.cs

```diff
@@ -22,10 +24,13 @@ internal class CSharpGen : IPlugin
         public async Task<GeneratedCodeWorkspace> ExecuteAsync(CodeModel codeModel)
         {
             ValidateConfiguration();
-
             Directory.CreateDirectory(Configuration.OutputFolder);
             var project = await GeneratedCodeWorkspace.Create(Configuration.AbsoluteProjectFolder, Configuration.OutputFolder, Configuration.SharedSourceFolders);
-            var sourceInputModel = new SourceInputModel(await project.GetCompilationAsync());
+
+            var baselineContract = await LoadBaselineContract();
+            var sourceInputModel = baselineContract is not null
+                ? new SourceInputModel(await project.GetCompilationAsync(), previousContract: baselineContract)
+                : new SourceInputModel(await project.GetCompilationAsync());
```

Comment:
could this just simply be
```suggestion
            var sourceInputModel = new SourceInputModel(await project.GetCompilationAsync(), previousContract: baselineContract);
```

Reply:
updated